lib/effects/BackgroundEffectProcessor.ts

```diff
@@ -0,0 +1,292 @@
+     */
+    async loadEffect(effect: BackgroundEffect) {
+        if (!this._tflite) {
+            log.debug('[loadEffect] loading tflite');
+            const modelToLoad = this._useWasm ? TFLITE_MODELS_SEG_LITE.model144 : TFLITE_MODELS_SEG_LITE.model96;
+            this._segmentationMaskDimensions = this._useWasm ? MODEL_WASM_INFERENCE_DIMENSIONS.model144 : MODEL_WASM_INFERENCE_DIMENSIONS.model96;
+            this._tflite = await loadTFLite(this._assetsPath + TFLITE_SIMD_LOADER_NAME, this._assetsPath + modelToLoad);
+        }
+        log.debug('[loadEffect] effect: ', effect.constructor.name);
+        this._effect = effect;
+        this._effect.init(this._tflite, this._segmentationMaskDimensions, this._inputVideoElement, this._outputCanvasCtx);
+    }
+
+
+    private set outputState(state: OutputState) {
+        if (state === this._outputState) return;
+        log.debug('[outputState] changed from', OutputState[this._outputState], 'to', OutputState[state]);
+        this._outputState = state;
+        this.applyOutputState();
+    }
+
+    private applyOutputState() {
+        // make sure to stop everything first before triggering new effects.
+        this.stopEffectProcessing();
+        if (this._inputForwardEffect) {
+            this._inputForwardEffect.stopDirectForwardingInputToOutput();
+            this._inputForwardEffect = null;
+        }
+
+        if (this._outputState === OutputState.EFFECT_APPLIED) {
+            log.debug('[applyOutputState] triggerEffectProcessing for background effect');
+            this.triggerEffectProcessing();
+        }
+
+        if (this._outputState === OutputState.INPUT_FORWARDING) {
+            if (supportsDirectForwarding()) {
+                log.debug('[applyOutputState] forwardInputToOutput');
+                this._inputForwardEffect = new VideoTrackToCanvas(this._inputVideoTrack, this._outputCanvasCtx);
+                this._inputForwardEffect.startDirectForwardingInputToOutput();
+            } else {
+                log.debug('[applyOutputState] triggerEffectProcessing for input forwarding');
+                this.triggerEffectProcessing();
+            }
+        }
+    }
+
+    private triggerEffectProcessing() {
+        this._maskFrameTimerWorker.postMessage({
+            id: SET_TIMEOUT,
+            timeMs: 1000 / this._outputFramesPerSecond
+        });
+    }
+
+    private stopEffectProcessing() {
+        this._maskFrameTimerWorker.postMessage({
+            id: CLEAR_TIMEOUT
+        });
+    }
+
+    /**
+     * Returns the previously set media stream
+     */
+    public get inputStream(): MediaStream {
+        return this._inputStream;
+    }
+
+    /**
+     * Sets / changes the input stream to the given stream. The effect will be applied to this stream and then forwarded to the ouput.
+     * @param stream
+     */
+    public set inputStream(stream: MediaStream) {
```

Comment:
I would prefer to keep the getter and setter simple and not to implement the logic into them. So I would suggest adding a function **setInputStream**(stream: MediaStream) which implements the logic and having the setter to just set the inputStream (`this._inputStream = stream;`).

Same thing for outputState and effectEnabled. Wdyt?

Reply:
I agree, I'll make the inputStream readonly then

Reply:
@enricop89 what do you think about the updated version?

Reply:
@davideberlein I don't see any new changes. Can you check if you have pushed them please?

Reply:
when I came back to my computer I saw that the push had failed, sorry.